### ISSUE TITLE
@BeforeClass and @AfterClass methods are still run even if no tests will be run.

### ISSUE THREAD

**Issue by noel-yap** [junit-team/junit#380]
PASSES:

```java
@Ignore
public class NotReallyATest {
    @BeforeClass
    public static void setUpClass() {
        fail("setUpClass() should not run");
    }

    @AfterClass
    public static void tearDownClass() {
        fail("tearDownClass() should not run");
    }

    @Test
    public void test() {
        fail("test() should not run");
    }
}
```

FAILS:

```java
public class NotReallyATest {
    @BeforeClass
    public static void setUpClass() {
        fail("setUpClass() should not run"); // fails here
    }

    @AfterClass
    public static void tearDownClass() {
        fail("tearDownClass() should not run"); // fails here
    }

    @Ignore
    @Test
    public void test() {
        fail("test() should not run");
    }
}
```

FAILS (using Request.filterWith() to filter out tests Categorized as DevMachine):

```java
@Category(DevMachine.class)
public class NotReallyATest {
    @BeforeClass
    public static void setUpClass() {
        fail("setUpClass() should not run"); // fails here
    }

    @AfterClass
    public static void tearDownClass() {
        fail("tearDownClass() should not run"); // fails here
    }

    @Test
    public void test() {
        fail("test() should not run");
    }
}
```

FAILS (using Request.filterWith() to filter out tests Categorized as DevMachine):

```java
public class NotReallyATest {
    @BeforeClass
    public static void setUpClass() {
        fail("setUpClass() should not run"); // fails here
    }

    @AfterClass
    public static void tearDownClass() {
        fail("tearDownClass() should not run"); // fails here
    }

    @Category(DevMachine.class)
    @Test
    public void test() {
        fail("test() should not run");
    }
}
```

**Comment by modocache** [comment]
This issue should probably be closed--#752 added tests that verify this bug was fixed a year ago! :calendar: :tada:

**Comment by stefanbirkner** [comment]
Fixed by #752. @modocache thanks for inspecting the issue.
